components/localized_input/localized_input.jsx

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {intlShape} from 'react-intl';
+
+export default class LocalizedInput extends React.PureComponent {
+    static propTypes = {
+
+        /**
+         * The placeholder text to render, has id and defaultMessage
+         */
+        placeholder: PropTypes.object.isRequired,
+    };
+
+    static contextTypes = {
+
+        /**
+         * react-intl helper object
+         */
+        intl: intlShape.isRequired,
+    };
+
+    get value() {
+        return this.input.value;
+    }
+
+    set value(value) {
+        this.input.value = value;
+    }
+
+    focus = () => {
+        this.input.focus();
+    };
+
+    setInput = (input) => {
+        this.input = input;
+    };
+
+    render() {
+        const {formatMessage} = this.context.intl;
+        const {placeholder, ...otherProps} = this.props;
+        let placeholderString = '';
+        if (placeholder.id) {
```

Comment:
For me, this must exist always, if not, you are not using correctly the component. I don't see the need of this "defensive" programming here, I would use directly `const placeholderString = formatMessage(placeholder);`